easybuild/tools/modules.py

```diff
@@ -718,6 +719,53 @@ def loaded_modules(self):
 
         return loaded_modules
 
+    def check_loaded_modules(self):
+        """
+        Check whether any (EasyBuild-generated) modules are loaded already in the current session
+        """
+        ignored_keys = [get_software_root_env_var_name(x) for x in build_option('allow_loaded_modules') or []]
+
+        eb_module_keys = []
+        for key in os.environ:
+            if key.startswith(ROOT_ENV_VAR_NAME_PREFIX) and key not in ignored_keys:
+                eb_module_keys.append(key)
+
+        if eb_module_keys:
+            loaded_modules = self.loaded_modules()
+
+            # try to track down modules that define the $EBROOT* environment variables that were found
+            loaded_eb_modules = []
+            for loaded_module in loaded_modules:
+                out = self.show(loaded_module)
+                for key in eb_module_keys[:]:
+                    if key in out:
+                        loaded_eb_modules.append(loaded_module)
+                        eb_module_keys.remove(key)
+
+            # warn about $EBROOT* environment variables without matching loaded module
+            if eb_module_keys:
+                tup = (len(eb_module_keys), ROOT_ENV_VAR_NAME_PREFIX, '$' + ', $'.join(eb_module_keys))
+                print_warning("Found %d defined $%s* environment variables without matching loaded module: %s" % tup)
```

Comment:
I'm not sure, who knows what's in the environment...
It's definitely worth warning about it though.

Also, I wouldn't be surprised if people already trick EB into doing this by defining `$EBROOT*` variables manually.

Which makes me really, I need to check how this plays together with the support for using external modules... I know we define `$EBROOT*` variables for external modules that get loaded (cfr. http://easybuild.readthedocs.io/en/latest/Using_external_modules.html#handling-of-provided-metadata), but I need to check where we do that exactly.